pkg/mq/interface.go

```diff
@@ -0,0 +1,163 @@
+package mq
+
+import "time"
+
+// Interface must be implemented for a message queuer
+type Interface interface {
+	// Connect to message queuer
+	Connect(timeout time.Duration) error
+
+	// Close connection
+	Close() error
+
+	// Publish
+	Publish(subject string, data []byte) error
+	PublishAsync(subject string, data []byte, ah AckHandler) (string, error)
+
+	// Subscribe
+	Subscribe(subject string, mh MsgHandler, opts ...SubscriptionOption) (Subscription, error)
```

Comment:
It's implemented here: https://github.com/appcelerator/amp/pull/710/files/74ea62ba211280cfcf4173a7eb13249e0684a260#diff-3d5bb591c85059b41e5cf90eef60af85R21